### PR TITLE
Solved Issue #4006

### PR DESCRIPTION
**Pull request by joshuazeltser**
Fixed issue #4006. To ensure focus returns to the "Edit Metadata" button after closing the popup, we stored the button once created (in JS) and then set it to focus on the button once the popup is hidden.

### CODE REVIEW

**Comment by takluyver** [comment]
NB the `edit_metadata` dialog function is used for both cell metadata and notebook metadata. Does this do anything strange when it's triggered by notebook metadata?

Also, are there any other ways that editing cell metadata can be triggered? E.g. can it be bound to a keyboard shortcut?

**Comment by joshuazeltser** [comment]
I have added a check to see whether the edit metadata is present before trying to focus on it. This should completely now solve the problem that this issue is addressing

File: notebook/static/base/js/dialog.js [hunk 1]
```diff
@@ -234,7 +234,8 @@ define(['jquery',
         });
 
         modal_obj.on('shown.bs.modal', function(){ editor.refresh(); });
-        modal_obj.on('hide.bs.modal', function(){options.edit_metadata_button.focus(); });
+        modal_obj.on('hide.bs.modal', function(){
+            options.edit_metadata_button ? options.edit_metadata_button.focus() : ""});
```
Comment (takluyver) [inline notebook/static/base/js/dialog.js]: Should there be a `;` inside the anonymous function?

(Nitpicking,  I know)

Comment (joshuazeltser) [reply notebook/static/base/js/dialog.js]: Done

**Pull request merged**